File: Sorting/BubbleSort.cpp
/*
 * Bubble Sort
 * 
 * example input: 5 2 4 1 6 3
 * 
 */

#include <iostream>

using namespace std;

int Bubble(int arr[],int size);
int swap(int a,int b);

int main(void)
{
	int array[6]={5,2,4,1,6,3};
	
	Bubble(array,6);
	
	return 0;
}

int Bubble(int arr[],int size)
{
	int i,j;
	
	for(i=0;i<size;i++)
	{
		for(j=0;j<size-1;j++)
		{
			if(arr[j]>arr[j+1])
			{
				swap(arr[j+1],arr[j]);
			}
		}
		cout<<"\n"<<i<<" : ";
		for(int k=0;k<size;k++)
			cout<<arr[k]<<" ";
	}
	
	cout<<"\n The sorted array is ";
	for(i=0;i<size;i++)
		cout<<arr[i]<<" ";
		
	return 0;
}


int swap(int a,int b)
{
	int temp;
	
	temp=a;
	a=b;
	b=temp;
	return 0;
}

File: Sorting/java/BubbleSort.java
import java.util.*;

public class BubbleSort {

  public static void log(int[] arr) {
    for(int i=0;i<arr.length;i++) {
      System.out.print(arr[i]+ " ");
    }
    System.out.println(" ");
  }

  public static int[] sort(int[] arr) {
    for(int i=arr.length-1;i>0;i--) {

      int end = arr.length - i;
      System.out.println("==SUB START==");
      for(int j=arr.length -1; j>=end; j--) {
        if (arr[j] < arr[j-1]){
          int temp = arr[j];
          arr[j] = arr[j-1];
          arr[j-1] = temp;
        }
        log(arr);
      }
      System.out.println("==SUB END==");
      System.out.println(" ===========START=========== ");
      log(arr);
      System.out.println(" ===========END=========== ");
    }
    return arr;
  }



  public static void main(String[] args) {
    int[] randomNumbers = {4,7,1,3,9,67,12,45,87,65,77,20};
    int[] sorted = sort(randomNumbers);
    System.out.println(sorted.length);
  }
}
